src/auth/auth.controller.ts

```diff
@@ -1,25 +1,25 @@
-import { Controller, Post, Body, Req } from '@nestjs/common';
+import { Controller, Post, Body, Req, Headers } from '@nestjs/common';
 import { AuthService } from './auth.service';
 import { ApiOperation, ApiTags } from '@nestjs/swagger';
 import { UserSignupDto } from './dto/user-signup.dto';
-import { Serialize } from '../common/interceptors/serialize.interceptor';
-import { SerializeUserDto } from './dto/serialize-user.dto';
 import { UserSigninDto } from './dto/user-signin.dto';
 import { Request } from 'express';
 import { Ok, ZaLaResponse } from 'src/common/helpers/response';
 import { userSignInType } from 'src/common/types';
+import { PasswordForgotEmailDto } from 'src/user/dto/password-email.dto';
+import { PasswordResetDto } from 'src/user/dto/password-reset.dto';
+import { User } from 'src/entities/user.entity';
 
 @ApiTags('Auth-Users')
 @Controller('auth')
 export class AuthController {
   constructor(private readonly authService: AuthService) {}
 
-  @Serialize(SerializeUserDto)
-  @Post('/signup')
+ @Post('/signup')
   @ApiOperation({ description: 'Sign up a User' })
   async signUpUser(@Body() body: UserSignupDto) {
     const user = await this.authService.signup(body);
-    return user;
```

Comment:
![pass](https://user-images.githubusercontent.com/38318744/187674480-9b42f3b2-4660-4e9f-bc65-522469c2eefb.png)

you are returning the entire user object which include the password, i dont think its necessary

Reply:
the frontend will know what to do with that.. i purposely did that because of the ZaLa Response object i wanted to use

Reply:
you can add serialization to limit what you send to the frontend relating to user